src/utils/communication_buffer.hpp

```diff
@@ -202,30 +230,32 @@ CommBuffer<T> &CommBuffer<T>::operator=(const CommBuffer<U> &in) {
   comm_ = in.comm_;
   active_ = in.active_;
   my_rank = Globals::my_rank;
+  get_resource_ = in.get_resource_;
   return *this;
 }
 
 template <class T>
-void CommBuffer<T>::Send() noexcept {
+void CommBuffer<T>::Send(bool local, int size) noexcept {
   if (!active_) {
-    SendNull();
+    SendNull(local);
     return;
   }
 
   PARTHENON_DEBUG_REQUIRE(*state_ == BufferState::stale,
                           "Trying to send from buffer that hasn't been staled.");
   *state_ = BufferState::sending;
-  if (*comm_type_ == BuffCommType::sender) {
+  if (*comm_type_ == BuffCommType::sender && !local) {
```

Comment:
Does this have implications on the legacy machinery (or not because `local` is default init to `false`)?

Reply:
As you say, because it is set to `false` by default it should have no impact on the legacy machinery.